New command: viva engage community add. Closes #5753 (#5976)
pnp/cli-microsoft365: docs/docs/cmd/viva/engage/engage-community-add.mdx

import Global from '/docs/cmd/_global.mdx';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# viva engage community add

Creates a new community in Viva Engage.

## Usage

```sh
m365 viva engage community add [options]
```

## Options

```md definition-list
`--displayName <displayName>`
: The name of the community. The maximum length is 255 characters.

`--description <description>`
: The description of the community. The maximum length is 1024 characters.

`--privacy <privacy>`

> : Defines the privacy level of the community. The possible values are: public, and private.

Reviewer: ```suggestion
: Defines the privacy level of the community. The possible values are: `public`, `private`.
```

`--adminEntraIds [adminEntraIds]`

> : Comma-separated list of Microsoft Entra IDs, assigning admin privileges to the designated individuals. Required when using application permissions. Specify either adminEntraIds or adminEntraUserNames, but not both.
> 
> `--adminEntraUserNames [adminEntraUserNames]`
> : Comma-separated list of Microsoft Entra UPNs, assigning admin privileges to the designated individuals. Required when using application permissions. Specify either adminEntraIds or adminEntraUserNames, but not both.

Reviewer: ```suggestion
: Comma-separated list of Microsoft Entra IDs, assigning admin privileges to the designated individuals. Required when using application permissions. Specify either `adminEntraIds` or `adminEntraUserNames`, but not both.

`--adminEntraUserNames [adminEntraUserNames]`
: Comma-separated list of Microsoft Entra UPNs, assigning admin privileges to the designated individuals. Required when using application permissions. Specify either `adminEntraIds` or `adminEntraUserNames`, but not both.
```

`--wait`
: Wait for the operation to finish.
```

<Global />

## Remarks

:::warning

> This command is based on an API that is currently in preview and is subject to change once the API reached general availability.

Reviewer: ```suggestion
This command is based on an API that is currently in preview and is subject to change once the API reaches general availability.
```

:::

> It is limited to networks in Native mode only - legacy and external Yammer networks will not be able to use this API for community creation.

Reviewer: ```suggestion
Creating a community is limited to networks in Native mode only - legacy and external Yammer networks will not be able to use this API for community creation.
```

Creating a community will create a linked Microsoft 365 unified group as well.

> In order to use this command, you need to grant the Microsoft Entra application used by the CLI for Microsoft 365 the permission to the Viva Engage API. To do this, execute the `cli consent --service VivaEngage` command.

Reviewer: For this command, this is not needed.

## Examples

Create a public community and wait for the community to be created

```sh
m365 viva engage community add --displayName "Software engineers" --description "A community for all software engineers" --privacy public --wait
```

Create a private community

```sh
m365 viva engage community add --displayName "Software engineers" --description "A community for all software engineers" --privacy private
```

Create a private community with owners

```sh
m365 viva engage community add --displayName "Software engineers" --description "A community for all software engineers" --privacy private --adminEntraUserNames "john.doe@contoso.onmicrosoft.com,jane.doe@contoso.onmicrosoft.com"
```

## Response

> ### With --wait:

Reviewer: When listing multiple responses, we always have a default response, and other responses. Check `planner plan set` as example.

  ```

  </TabItem>
  <TabItem value="Markdown">

  ```md
  # viva engage community add --displayName "Software engineers" --description "A community for all software engineers" --privacy "private" --wait "true"

  Date: 10/04/2024

  ## eyJfdHlwZSI6IkxvbmdSdW5uaW5nT3BlcmF0aW9uIiwiaWQiOiI5ZTgzMDRjMC1mZmY0LTRkODgtYTM4OS1mNDY0MmQ3OGZlMTciLCJvcGVyYXRpb24iOiJDcmVhdGVDb21tdW5pdHkifQ

  Property | Value
  ---------|-------
  id | eyJfdHlwZSI6IkxvbmdSdW5uaW5nT3BlcmF0aW9uIiwiaWQiOiI5ZTgzMDRjMC1mZmY0LTRkODgtYTM4OS1mNDY0MmQ3OGZlMTciLCJvcGVyYXRpb24iOiJDcmVhdGVDb21tdW5pdHkifQ
  createdDateTime | 2024-04-10T12:47:47.3488087Z
  lastActionDateTime | 2024-04-10T12:47:47.3488107Z
  status | succeeded
  resourceLocation | https://graph.microsoft.com/beta/employeeExperience/communities('eyJfdHlwZSI6Ikdyb3VwIiwiaWQiOiIxODQ4MTUyMzkxNjgifQ')
  operationType | createCommunity
  resourceId | eyJfdHlwZSI6Ikdyb3VwIiwiaWQiOiIxODQ4MTUyMzkxNjgifQ
  ```

  </TabItem>
</Tabs>

### Without --wait:

<Tabs>
  <TabItem value="JSON">

  ```json

>   https://graph.microsoft.com/beta/employeeExperience/engagementAsyncOperations('eyJfdHlwZSI6IkxvbmdSdW5uaW5nT3BlcmF0aW9uIiwiaWQiOiI4ZmM2NzEyZS0wMWY4LTQxN2YtYWNmMS1iZTJiYmMxY2FjNGQiLCJvcGVyYXRpb24iOiJDcmVhdGVDb21tdW5pdHkifQ')

Reviewer: When output is JSON, this should be a JSON string. So we're missing quotes around this string.
